### PR TITLE
[13.0][FIX] Do not send disallowed chars on postlogistics api

### PR DESCRIPTION
**Pull request by mmequignon**
Some chars are disallowed from postlogistics API ("|", "\", "<", ">") and output json should be sanitized

Based on https://github.com/OCA/delivery-carrier/pull/367.
Those will need to be rebased when one will be merged.

### CODE REVIEW

File: delivery_postlogistics/tests/test_sanitize_values.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                "delivery_type": "postlogistics",
+                "product_id": shipping_product.id,
+                "postlogistics_endpoint_url": ENDPOINT_URL,
+                "postlogistics_client_id": CLIENT_ID,
+                "postlogistics_client_secret": CLIENT_SECRET,
+                "postlogistics_label_layout": label_layout.id,
+                "postlogistics_output_format": output_format.id,
+                "postlogistics_resolution": image_resolution.id,
+            }
+        )
```
Comment (sebalix) [inline delivery_postlogistics/tests/test_sanitize_values.py]: All this code is the same than the one in `test_postlogistics.py` (as well as `ENDPOINT_URL` etc), it deserves a common class both tests should inherit from.

**Review by sebalix — APPROVED** [review]

**Comment by sebalix** [comment]
Depends on https://github.com/OCA/delivery-carrier/pull/342 to get the refactoring of tests, waiting for it to be merged.

**Review by simahawk — APPROVED** [review]

File: delivery_postlogistics/postlogistics/web_service.py [hunk 2]
```diff
@@ -434,6 +445,13 @@ def get_access_token(cls, picking_carrier):
             cls.access_token_expiry = now + timedelta(seconds=response["expires_in"])
             return cls.access_token
 
+    def _sanitize_string(self, value):
+        """Removes disallowed chars ("|", "\", "<", ">") from strings."""
+        if isinstance(value, str):
```
Comment (gurneyalex) [inline delivery_postlogistics/postlogistics/web_service.py:453]: if we have False in there, do we need to return False or an empty string?

Comment (mmequignon) [reply delivery_postlogistics/postlogistics/web_service.py:453]: The idea is to remove  disallowed chars from incoming strings, I don't think we should alter the value if it's something else than a string. That case should already be handled by the base implementation.

Comment (mmequignon) [reply delivery_postlogistics/postlogistics/web_service.py:453]: I mean, sending `{"whatever": null}` and `{"whatever": ""}` is not the same, and I don't want to modify something that is currently working.

**Comment by simahawk** [comment]
@mmequignon pls rebase and drop the 1st commit

**Pull request merged**